src/odemis/driver/technolution.py

```diff
@@ -1510,12 +1520,13 @@ def system_checks(self, ASM):
         :param ASM (AcquisitionServer object) : AcquisitionServer object used for changed the state of the system and
         checking system parameters.
         """
-        # Try statements are used to prevent triggering an error in an error.
+        # Try statements are implemented to prevent triggering an error in an error which would cascade infinite errors.
```

Comment:
When will the try statement fail (the asmApiGetCall)?

If the asmApiGetCall fails already at the `_session.get()` call, you get a session error I believe. 
If it fails because of the `resp.status_code` and `expected_status` not matching you get a AsmApiException. 
Otherwise you get a response returned (which is True or False). 

What is the `cascade of infinite errors`?

Reply:
These systems checks are done within the error class AsmApiException, these are here to give extra info if an error occurs.
The try statements could fail if any of the system_checks calls do not return the expected result (aka another response than 200).
The try statments are there to prevent recursivly creating errors, this can happen if for example a call to the ext_storage will cause an error. This error will be created in the AsmApiException, which will (without the try statements) cause and error when checking the state of the ext_storage. This new error is created via the AsmApiException class which will again check the ext_storage and create the cascade of errors.
To prevent this the try statements are used. This way extra information can be presented to the user if the systems checks won't raise any error. But no infinite cascase will be created.

Reply:
It might then not the best solution that the system checks done in the AsmApiException also raise again an AsmApiException. Maybe Eric has a good idea how that should be done or if it is ok to do it like it is now.

Reply:
The current implementation works and should not cause infinite errors.
It is ok to do it like it implemented now because of the Try statements, these staments prevent the infinite loop of errors.